Fixed a few issues with Naomi 2.2

aaronchantrill opened the pull request:
## Description
Added the generated Naomi file that uses the local python to the
.gitignore list

Modified naomi-setup.sh and added libncurses5-dev to the list
of apt packages to install. Also fixed a bug that was preventing
the "$@" from writing to the end of the ./Naomi file, so it was
impossible to forward command line arguements to ./Naomi.py

Removed nano and git from python-requirements.txt (duplicating
PR # 131 by redragonx)

Added a check for audio engines to populate.py that runs directly
after the language selection, so the user doesn't get to the end
and suddenly discover they can't complete the install.

Fixed an issue with sphinxvocab.py where temporary files were
not being removed. This was for debugging purposes.

## Related Issue
[#136 Naomi-setup.sh issues](https://github.com/NaomiProject/Naomi/issues/136)

## Motivation and Context


## How Has This Been Tested?
Tested on a clean install of Raspbian 2018-11-13

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

aaronchantrill committed:
Fixed a few issues with Naomi 2.2

Added the generated Naomi file that uses the local python to the
.gitignore list

Modified naomi-setup.sh and added libncurses5-dev to the list
of apt packages to install

Removed nano and git from python-requirements.txt (duplicating
PR # 131 by redragonx)

Added a check for audio engines to populate.py that runs directly
after the language selection, so the user doesn't get to the end
and suddenly discover they can't complete the install.

Fixed an issue with sphinxvocab.py where temporary files were
not being removed. This was for debugging purposes.

AustinCasteel reviewed (commented):
The changes to the files on their own look fine. However in running tests I got two things that came up. [Pop error could just be because I did not do the sphinx config before running setup. However the setup traceback could be an issue.](https://gist.github.com/Longshotpro2/5b81abdc1afd8fdd9eaca0c462204c4e)

aaronchantrill committed:
More fixes for Naomi setup

Changed downloaded setuptools version to latest

Changed plugin search path for audiotools so it
no longer generates warnings from other plugins

AustinCasteel reviewed (approved):
Traceback fixed and verified working on RPI!
